test/WireMock.Net.Tests/Matchers/JsonPartialMatcherTests.cs

```diff
@@ -281,6 +281,25 @@ public void JsonPartialMatcher_IsMatch_JObjectAsString()
         Assert.Equal(1.0, match);
     }
 
+    [Fact]
+    public void JsonPartialMatcher_IsMatch_GuidAsString()
```

Comment:
`new { Id = 1, Name = "^1111[a-fA-F0-9]{4}(-[a-fA-F0-9]{4}){3}-[a-fA-F0-9]{12}$", false, false, true }`
**Name here has string type, not Guid.**
So value.Type will be JTokenType.String and condition on line 66 will be true.
We still can have input.Type ==  JTokenType.Guid but it doesn't matter, because it casts to string on line 70.
I've just checked - your test case is success even without any logic changed.

**This PR fixes the situation when Name is a Guid type** (and input.Type ==  JTokenType.Guid), not guid-like string which is possible only when you assign, for example, Guid.NewGuid() or Guid.Parse() to a field in anonymous object

Reply:
I've added the unit test based on your example, just to be sure that everything works correctly